python/tank/descriptor/io_descriptor/base.py

```
@@ -211,93 +212,94 @@ def _find_latest_tag_by_pattern(self, version_numbers, pattern):
-        version_split = re.findall("([0-9]+|x)", pattern)
-        if 'x' in version_split:
-            # check that we don't have an incorrect pattern using x
-            # then a digit, eg. v4.x.2
-            if re.match("^v[0-9\.]+[x\.]+[0-9\.]+$", pattern):
-                raise TankDescriptorError(
-                    "Incorrect version pattern '%s'. "
-                    "There should be no digit after a 'x'." % pattern
-                )
-
-        current = versions
-        version_to_use = None
-        # process each digit in the pattern
-        for version_digit in version_split:
-            if version_digit == 'x':
-                # replace the 'x' by the latest at this level
-                version_digit = max(current.keys(), key=int)
-            version_digit = int(version_digit)
-            if version_digit not in current:
-                # no matches
+        # Validate the pattern: we expect at least v<digit or x>.<digit or x>.<digit or x>
+        # - we can then have more .<digit or x>
+        # - we can then have a -<topic name>, optionally followed by multiple .<alphanumeric or _>
+        if not re.match("^v(\d+|x)(\.(\d+|x)){2,}(-\w+(\.\w+)*)?$", pattern):
+            raise TankDescriptorError(
+                "Cannot parse version expression '%s', it must follow a vx.x.x[-topic] pattern "
+                "where 'x' can either be a digit or the letter 'x', and 'topic' is an "
+                "optional arbitrary string." % pattern
+            )
+        # Split the pattern either with "." or "-", keep the separator in the result
+        # with the grouping in the regexp
+        version_split = re.split("(\.|-)", pattern)
+        num_tokens = len(version_split)
+        # Special case for the first token for which the wildcard is "vx" and not
+        # "x"
+        has_v_prefix = False
+        version_regexp = version_split[0]
+        if version_split[0] == "vx":
```

The prefix is indeed `v`. From `vx.x.x` we only consider `x.x.x`, which then match any version with the `v*.*.*` pattern. For the record, we could always assume a `v` prefix is used, as this is enforced when publishing in the app store and seems to be our convention.  So we could raise an error if a given pattern does not start with `v`. But I kept the feature that was previously implemented.